media-libs/osl/files/osl-1.10.5-find-openexr.patch

```diff
@@ -0,0 +1,120 @@
+From c0b174a226c03efd54c8d09ab6305101f7c4b00f Mon Sep 17 00:00:00 2001
+From: Bernd Waibel <waebbl@gmail.com>
+Date: Mon, 10 Jun 2019 12:52:35 +0200
+Subject: [PATCH] Fix to find openexr version information on a multilib system.
+
+The patch consequently uses pkg-config to not only locate the openexr
+headers, but additionally to get the openexr version.
+
+On Gentoo multilib systems (x86_64, x86) the headers of openexr are wrapped
+and the actual headers are installed somewhere under the gcc compilers
+include directory, so the using the cmake strings command fails to properly
+get the version information.
+
+Signed-off-by: Bernd Waibel <waebbl@gmail.com>
+---
+ src/cmake/modules/FindOpenEXR.cmake | 74 ++++++++++++++++-------------
+ 1 file changed, 41 insertions(+), 33 deletions(-)
+
+diff --git a/src/cmake/modules/FindOpenEXR.cmake b/src/cmake/modules/FindOpenEXR.cmake
+index 6c6b39c..6861dba 100644
+--- a/src/cmake/modules/FindOpenEXR.cmake
++++ b/src/cmake/modules/FindOpenEXR.cmake
+@@ -27,53 +27,61 @@ find_package(PkgConfig)
+ if (PKG_CONFIG_FOUND)
+     pkg_check_modules(_ILMBASE QUIET IlmBase>=2.0.0)
+     pkg_check_modules(_OPENEXR QUIET OpenEXR>=2.0.0)
++    if (_ILMBASE_FOUND AND _OPENEXR_FOUND)
++    	set (ILMBASE_INCLUDE_PATH ${_ILMBASE_INCLUDE_DIRS})
++    	set (OPENEXR_INCLUDE_PATH ${_OPENEXR_INCLUDE_DIRS})
++    	set (OPENEXR_VERSION ${_OPENEXR_VERSION})
++    	string (REGEX MATCH "[0-9]+" OPENEXR_VERSION_MAJOR ${OPENEXR_VERSION})
++    	string (SUBSTRING OPENEXR_VERSION 3 -1 _OPENEXR_VERSION_REMAINDER)
++    	string (REGEX MATCH "[0-9]+" OPENEXR_VERSION_MINOR ${_OPENEXR_VERSION_REMAINDER})
++    endif (_ILMBASE_FOUND AND _OPENEXR_FOUND)
+ endif (PKG_CONFIG_FOUND)
+ 
+ # List of likely places to find the headers -- note priority override of
+ # ${OPENEXR_ROOT_DIR}/include.
+ # ILMBASE is needed in case ilmbase an openexr are installed in separate
+ # directories, like NixOS does
+-set (GENERIC_INCLUDE_PATHS
+-    ${OPENEXR_ROOT_DIR}/include
+-    $ENV{OPENEXR_ROOT_DIR}/include
+-    ${ILMBASE_ROOT_DIR}/include
+-    $ENV{ILMBASE_ROOT_DIR}/include
+-    ${_ILMBASE_INCLUDEDIR}
+-    ${_OPENEXR_INCLUDEDIR}
+-    /usr/local/include
+-    /usr/include
+-    /usr/include/${CMAKE_LIBRARY_ARCHITECTURE}
+-    /sw/include
+-    /opt/local/include )
++#set (GENERIC_INCLUDE_PATHS
```

Comment:
did you comment these lines in your upstream patch as well? I'd consider them obsolete otherwise, and upstream likely as well.

Reply:
Yes, I commented them. IIRC at least some of those paths were introduced by upstream not long ago, so they might have customers who still use some machines using them.
I'd thought about deleting those lines completely instead of just commenting them. But for the purpose of documenting I decided to leave them and let upstream decide whether to remove them, if the patch gets accepted.
I planned to put them in the upstream patch as well. Haven't opened an issue, as I wanted to ensure, the patches don't need further changes.